src/extension.ts

```diff
@@ -29,43 +41,94 @@ const extract = async (options: {
     )
 
     const text = document.getText()
-    const { unbound, styledImported } = collectUnbound(text)
+    const unbound = collectUnbound(text)
     if (!unbound.length) {
       vscode.window.showWarningMessage(
-        '[SCE] Nothing to copy: There is no unbound components',
+        '[SCE] Nothing to extract: There are no unbound components',
       )
       return
     }
+
+    const exportIdentifier =
+      type == 'extractExportedToClipboard' || type == 'extractToSeparateFile'
+
     const declarations = generateDeclarations({
       unbound,
-      styledImported,
-      exportIdentifier: options.exportIdentifier,
-      importStyled: config.get('addImportStatement', true),
+      exportIdentifier,
     })
 
-    await vscode.env.clipboard.writeText(declarations)
-    vscode.window.showInformationMessage(
-      `[SCE] Copied to clipboard! (Found: ${unbound.length}) `,
-    )
+    if (type == 'extractToClipboard' || type == 'extractExportedToClipboard') {
+      let clipboardText = declarations
+      if (config.get('addImportStatement', true)) {
+        const styledImportInsertion = getStyledImportInsertion(
+          editor.document.getText(),
+        )
+        if (styledImportInsertion) {
+          clipboardText = styledImportInsertion.insertionText + declarations
+        }
+      }
```

Comment:
New handling of imports for clipboard case. The detection of whether to add imports is still a combination of the config and whether `styled` is imported already. The latter case is handled by `getStyledImportInsertion` which will return null if `styled` is imported already.